app/models/manageiq/providers/workflows/automation_manager/workflow_instance.rb

```diff
@@ -1,14 +1,20 @@
 class ManageIQ::Providers::Workflows::AutomationManager::WorkflowInstance < ManageIQ::Providers::EmbeddedAutomationManager::ConfigurationScript
-  def run_queue(zone: nil, role: "automation")
+  def run_queue(zone: nil, role: "automation", object: nil)
     raise _("run_queue is not enabled") unless Settings.prototype.ems_workflows.enabled
 
+    args = {:zone => zone, :role => role}
+    if object
+      args[:object_type] = object.class.name
+      args[:object_id]   = object.id
+    end
+
     queue_opts = {
       :class_name  => self.class.name,
       :instance_id => id,
       :method_name => "run",
       :role        => role,
       :zone        => zone,
-      :args        => [{:zone => zone, :role => role}],
+      :args        => [args],
```

Comment:
I don't understand the zone and role being in the args, when they're already in the queue item

Reply:
Oh it's cause run calls run_queue ? (that's really backwards with respect to our method naming - usually run_queue calls run but on the queue)

Reply:
It is because we have to put next state on the queue, so run_queue does put run on the queue but run can "requeue"

Reply:
ohhhh - ok - it's still weird but I get it now.